### PR TITLE
Add ADK field ionization to FLYonPIC

### PR DESCRIPTION
**Pull request by BrianMarre**
adds ADK based field ionization to FLYonPIC/AtomicPhysics

- [x] rebase against https://github.com/ComputationalRadiationPhysics/picongpu/pull/5081 before merging

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@BrianMarre I stopped all running jobs because one chain already failed so other will failing too.

https://gitlab.com/hzdr/crp/picongpu/-/pipelines/1447918284

**Comment by psychocoderHPC** [comment]
I added a point that the PR must be rebased to avoid that this PR is accessing old units e.g. DELTA_T, ...

File: include/picongpu/particles/atomicPhysics/enums/ChooseTransitionGroupFor.hpp [hunk 1]
```diff
@@ -45,12 +45,7 @@ namespace picongpu::particles::atomicPhysics::enums
         static constexpr ChooseTransitionGroup chooseTransitionGroup = ChooseTransitionGroup::boundBoundDownward;
     };
 
-    //! bound-free(upward)
-    template<>
-    struct ChooseTransitionGroupFor<TransitionType::boundFree, TransitionDirection::upward>
-    {
-        static constexpr ChooseTransitionGroup chooseTransitionGroup = ChooseTransitionGroup::boundFreeUpward;
-    };
+    //! bound-free(upward) may be either collisionalBoundFreeUpward or fieldBoundFreeUpward -> error to ask
```
Comment (psychocoderHPC) [inline include/picongpu/particles/atomicPhysics/enums/ChooseTransitionGroupFor.hpp]: what does `error to ask` mean?

File: include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel [hunk 2]
```diff
@@ -137,30 +143,38 @@ namespace picongpu::particles::atomicPhysics::kernel
...
                             }
                         }
                     }
+
                     // noChange transition as last resort transition, never stored in RateCache
-                    if(r >= cumSum) // and <= 1
+                    /// @todo test for no change first, since with default setting ~90% will take this branch anyway
+                    if(r >= cumSum) // and <= 1 by definition of timeStep
                     {
+                        // printf("ID: %lu, %i\n", ion[particleId_], u32(s_enums::ProcessClass::noChange));
```
Comment (psychocoderHPC) [inline include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel]: please remove, dead code

File: include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel [hunk 3]
```diff
@@ -137,30 +143,38 @@ namespace picongpu::particles::atomicPhysics::kernel
...
+                                ion[transitionIndex_] = chooseTransitionGroupID;
                                 break;
                             }
                         }
                     }
+
                     // noChange transition as last resort transition, never stored in RateCache
-                    if(r >= cumSum) // and <= 1
+                    /// @todo test for no change first, since with default setting ~90% will take this branch anyway
+                    if(r >= cumSum) // and <= 1 by definition of timeStep
```
Comment (psychocoderHPC) [inline include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel]: please move comment above

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]
minor changes required than it can go into mainline

File: include/picongpu/particles/atomicPhysics/enums/ChooseTransitionGroupFor.hpp (hunk 1 above)
Comment (BrianMarre) [reply include/picongpu/particles/atomicPhysics/enums/ChooseTransitionGroupFor.hpp]: discussed offline, moved comment to file documentation and extended the file description to make it better understandable what the intention of the file is.

File: include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel (hunk 3 above)
Comment (BrianMarre) [reply include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel]: fixed

File: include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel (hunk 2 above)
Comment (BrianMarre) [reply include/picongpu/particles/atomicPhysics/kernel/ChooseTransitionGroup.kernel]: fixed

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**